File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/INavigationControllerListener.java
package io.github.romangolovanov.apps.ametro.ui.navigation;


import io.github.romangolovanov.apps.ametro.model.entities.MapDelay;

public interface INavigationControllerListener {
    boolean onOpenMaps();

    boolean onOpenSettings();

    boolean onChangeScheme(String schemeName);

    boolean onToggleTransport(String source, boolean checked);

    boolean onDelayChanged(MapDelay delay);

    boolean onOpenAbout();
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/INavigationControllerListener.kt
package io.github.romangolovanov.apps.ametro.ui.navigation

import io.github.romangolovanov.apps.ametro.model.entities.MapDelay

interface INavigationControllerListener {
    fun onOpenMaps(): Boolean
    fun onOpenSettings(): Boolean
    fun onChangeScheme(schemeName: String): Boolean
    fun onToggleTransport(source: String, checked: Boolean): Boolean
    fun onDelayChanged(delay: MapDelay): Boolean
    fun onOpenAbout(): Boolean
}
